### PR TITLE
Adding default drop route before creating all other routes

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: orchagent/routeorch.cpp [hunk 1]
```diff
@@ -144,7 +175,15 @@ void RouteOrch::decreaseNextHopRefCount(IpAddresses ipAddresses)
 {
     if (ipAddresses.getSize() == 1)
     {
-        IpAddress ip_address(ipAddresses.to_string());
+        string ip_str = ipAddresses.to_string();
```
Comment (stcheng) [inline orchagent/routeorch.cpp]: no need to have an extra variable?

File: orchagent/routeorch.cpp [hunk 2]
```diff
@@ -10,6 +10,37 @@ extern sai_route_api_t*             sai_route_api;
...
+    attr.id = SAI_ROUTE_ATTR_PACKET_ACTION;
+    attr.value.s32 = SAI_PACKET_ACTION_DROP;
+
+    sai_status_t status = sai_route_api->create_route(&unicast_route_entry, 1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create default route with packet action drop");
+    }
+
+    // add default route into the m_syncdRoutes
```
Comment (qiluo-msft) [inline orchagent/routeorch.cpp]: should below statements be moved into if-block?

File: orchagent/routeorch.cpp [hunk 3]
```diff
@@ -144,7 +175,15 @@ void RouteOrch::decreaseNextHopRefCount(IpAddresses ipAddresses)
 {
     if (ipAddresses.getSize() == 1)
     {
-        IpAddress ip_address(ipAddresses.to_string());
+        string ip_str = ipAddresses.to_string();
+
+        // skip blackhole route
+        if (ip_str == "0.0.0.0")
```
Comment (qiluo-msft) [inline orchagent/routeorch.cpp]: Parse it as ipv4 and compared to all zero? I guess "00.0.0.000" is valid IP.

File: orchagent/routeorch.cpp (hunk 1 above)
Comment (lguohan) [inline orchagent/routeorch.cpp]: two places below use ip_str, do not want to do conversion two times.

File: orchagent/routeorch.cpp [hunk 4]
```diff
@@ -417,11 +456,28 @@ bool RouteOrch::removeRoute(IpPrefix ipPrefix)
     route_entry.vr_id = gVirtualRouterId;
     copy(route_entry.destination, ipPrefix);
 
-    sai_status_t status = sai_route_api->remove_route(&route_entry);
-    if (status != SAI_STATUS_SUCCESS)
+    // set to blackhole for default route
+    if (ipPrefix.to_string() == "0.0.0.0/0")
```
Comment (qiluo-msft) [inline orchagent/routeorch.cpp]: Parse and compare?

File: orchagent/routeorch.cpp (hunk 2 above)
Comment (lguohan) [inline orchagent/routeorch.cpp]: throw exceptions

File: orchagent/routeorch.cpp (hunk 1 above)
Comment (lguohan) [inline orchagent/routeorch.cpp]: updated

File: orchagent/routeorch.cpp (hunk 4 above)
Comment (lguohan) [inline orchagent/routeorch.cpp]: updated.

**Comment by lguohan** [comment]
updated

**Comment by qiluo-msft** [comment]
:shipit:

**Pull request merged**